userguide/analytics/userguide_analytics.adoc

```diff
@@ -1,63 +1,595 @@
+
+[[installing_canned_reports]]
+=== Installing Canned Reports
+
+As explained earlier, we provide a set of canned reports that can be useful as a starting point. On Non-Windows machines, these can be installed by running the https://github.com/killbill/killbill-analytics-plugin/blob/59c282f97960dc59b944ca33e17353bc19d7c842/src/main/resources/seed_reports.sh[seed_reports.sh] script. On Windows machines, you can install a report by opening the desired report sub-directory from the https://github.com/killbill/killbill-analytics-plugin/tree/master/src/main/resources/reports[reports] directory and creating the _view_, _stored procedure_ and _report_ as explained in the next section.
+
+[[create_custom_reports]]
+=== Creating Custom Reports
+
+As mentioned earlier, the analytics plugin allows creating custom reports as per your requirements. A custom report can be created by executing API endpoints or via Kaui. This section covers report creation via API endpoint execution.
+
+There are basically three steps in creating a report:
+
+. Create a table/view corresponding to the report.
++
+Suppose you want to view the number of accounts created per day. This is typically a <<timeline, timeline>> report and needs to include the `tenant_record_id` and `day` columns.
++
+.. Identify the SQL query corresponding to the report:
++
+[source,sql]
+----
+select
+  tenant_record_id
+, date_format(created_date, '%Y-%m-%d') as day
+, count(*) as count
+from analytics_accounts
+where report_group = 'default'
+group by 1,2
+order by 1,2 asc
+;
+----
+
++
+.. Create a view corresponding to this query as follows:
++
+[source,sql]
+----
+create or replace view v_new_accounts_per_day as
+select
+  tenant_record_id
+, date_format(created_date, '%Y-%m-%d') as day
+, count(*) as count
+from analytics_accounts
+where report_group = 'default'
+group by 1,2
+order by 1,2 asc
+;
+----
++
+. Create the actual report.
++
+You can then create the actual report by executing the following endpoint:
++
+[source,bash]
+----
+curl -v \
+     -X POST \
+     -u admin:password \
+     -H "X-Killbill-ApiKey:bob" \
+     -H "X-Killbill-ApiSecret:lazar" \
+     -H 'Content-Type: application/json' \
+     -d '{"reportName": "new_accounts_per_day",
+          "reportPrettyName": "New accounts created (per day)",
+          "sourceTableName": "v_new_accounts_per_day"}' \
+     "http://127.0.0.1:8080/plugins/killbill-analytics/reports"
+----
++
+This creates a report called `new_accounts_per_day` based on the `v_new_accounts_per_day` view. Refer to the <<Report Creation Options,Report Creation Options>> section for a description of all the fields.
++
+Alternatively, you can directly specify an SQL query (instead of creating a view) while creating the report. Refer to the <<advanced_report_creation_options, Advanced Report Creation>> section for further information.
```

Comment:
```suggestion
Alternatively, you can directly specify an SQL query (instead of creating a view) while creating the report. Refer to the <<advanced_report_creation_options, "Advanced Report Creation">> section for further information.
```

Reply:
Fixed